pkg/proc/proc.go

```diff
@@ -140,6 +120,7 @@ func (dbp *Process) Detach(kill bool) (err error) {
 			err = killProcess(dbp.pid)
 		}
 	})
+	dbp.Close()
```

Comment:
We could change target.Interface to have a method `BinaryInfo() *proc.BinaryInfo` instead, since this describes the executable it should be the same for all implementation and could be represented by a concrete type.